### DISCUSSION TITLE
[useSWRInfinite]  fetcher function not receiving updated values.

### DISCUSSION THREAD

**Discussion by shinygit** [vercel/swr#609] [General]
The issue I am having is every call of size + 1 while resulting in a new query does not change any of the values sent to the fetcher.  I am perhaps using it in a bit unusual of a way.  However I believe I also tried it correctly in a more conventional approach and could not get it to work as well.  I am sure I am missing something.
```
const [searchCreditsQuery, setSearchCreditsQuery] = useState({
    searchTerm: '',
    claimed: false,
  })
  const getKey = (pageIndex, previousPageData) => {
    if (previousPageData && !previousPageData.length) return null
    return [
      `/api/credits/${teamid}/search`,
      searchCreditsQuery.searchTerm,
      searchCreditsQuery.claimed,
      size,
    ]
  }
  const {
    data: creditsData,
    error: creditsError,
    isValidating,
    mutate,
    size,
    setSize,
  } = useSWRInfinite(getKey, () =>
    searchCredits(teamid, {... searchCreditsQuery, skip: size * 10})
  )
```
Whenever I call setSize(size => size+1), searchCredits only receives skip: 10 no matter how many times.
size is increasing with each call but its not passed down.
Any thoughts or guidance would be appreciated.

**Comment by shinygit** [comment]
Seems like whatever I do any variables passed to the fetcher function don't update when they change.  Perhaps I am just misunderstanding javascript.

**Reply by srodriguezbf** [reply]
Where you able to work around this? Facing same issue

**Reply by jgoedde** [reply]
Same issue here...
